Cassandra set tests

conorbros opened the pull request.

conorbros commented:
Update: Trying to find the cause of this failing test, only happens intermittently on my machine.

rukai commented:
its happening on main branch as well.
I havent been able to reproduce it on my machine.

conorbros commented:
Doesn't happen when I isolate the `varint` test either...fun!

rukai commented:
Oh wait, its failing on the test you changed. Ignore what I said.

rukai commented:
![image](https://user-images.githubusercontent.com/5120858/149859319-177bd49e-6d18-4be5-b0c9-c72ea5121815.png)
maybe you just forgot to implement conversion from Varint to i64?

conorbros commented:
It sometimes passes on my local machine though

rukai commented:
Maybe something is failing silently and skipping part of the test under some circumstances?

conorbros commented:
Alright I've fixed all the issues it seems. Changes: 
- Upstreamed some fixes to cdrs-tokio.
- Increased heap memory for Cassandra. Was running out of memory due to all the data/migrations the test suite does. 
- removed the `duration` type test because it's not actually a native type in Cassandra and is instead part of the custom types.

rukai commented:
Nice summary of the problems.
I'll take a look now.

rukai commented on shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:
```diff
@@ -557,10 +573,176 @@ mod collections {
+        }
+
+        fn insert(session: &Session) {
+            // insert lists of native types
```
outdated comment?

rukai commented on shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:
```diff
@@ -557,10 +573,176 @@ mod collections {
+                );
+            }
+
+            fn get_map_example(_key: usize, value: &str) -> String {
```
can we deduplicate this?

rukai commented on shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:
```diff
@@ -557,10 +573,176 @@ mod collections {
+                );
+            }
+
+            // test selecting set  of maps
```
double space

rukai reviewed (changes requested):
just a few minor changes

conorbros commented on shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:
```diff
@@ -557,10 +573,176 @@ mod collections {
+                        get_type_str(*native_col_type)
+                    )
+                    .as_str(),
+                );
```
Ended up fixing a couple of the other lines in this file aswell

rukai reviewed (approved):
LGTM, glad we found the cause of the random failures on main.
